### PR TITLE
Ramp board/leave speedometer

### LINKED ISSUE #637: Ramp enter and exit speed on speedometer
**What feature is your improvement idea related to? Please describe.**
Suggested by Sokaries [back in October 2019](https://i.imgur.com/cGezg93.png):
"Having ramp entry/leave speed somewhere in HUD would be great"

**Describe the solution you'd like**
For now it can be two extra labels in the speedometer, fetching the speed values from the player class (it'll be networked values on the player).

Eventually it will be good to figure out if it should be like the last jump velocity label and show temporarily and fade out as you get them.

### PR DESCRIPTION
**Pull request by braem**
Closes #637 

Added a new speedometer label that shows both the ramp enter and leave speed. Kept them in the same label as the comparison coloring can compare between board and leave this way.

Did this by using @Gocnak 's code for ramp board & leave, then created events for them so the speedometer could capture & display them. I passed in the values through the event rather than getting the player's velocity during capture, not sure which direction would be best for that but it is an easy change.

Added to hud settings page. Had to increase the combobox width to fit it in.
image

Also, while I was in the area, I cleaned up the stage enter/exit code as it was getting messy.

Will attach a video playing with it later tonight if curious.

### Checklist
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

**Comment by braem** [comment]
@Gocnak : in order to rebase I needed to edit your commit to remove the speedometer adding stuff (just keeping the changes in player and gamemovement), but in doing so I'm listed as the author in the commit which feels wrong.

File: mp/src/game/shared/momentum/mom_gamemovement.cpp [hunk 1]
```diff
@@ -2237,8 +2267,26 @@ int CMomentumGameMovement::TryPlayerMove(Vector *pFirstDest, trace_t *pFirstTrac
...
+                        //engine->Con_NPrintf(2, "Ramp board: %f", mv->m_vecVelocity.Length());
+                        m_pPlayer->m_vecRampBoardVel = mv->m_vecVelocity;
+                        m_pPlayer->m_bSurfing = true;
+
+                        IGameEvent *pEvent = gameeventmanager->CreateEvent("ramp_board");
+                        if (pEvent)
+                        {
+                            pEvent->SetFloat("speed", mv->m_vecVelocity.Length());
+                            gameeventmanager->FireEvent(pEvent);
+                        }
```
[the remark this answers is not in the document]
Comment (braem) [reply mp/src/game/shared/momentum/mom_gamemovement.cpp:2285]: as in `SetRampBoardVelocity` and `GetRampBoardVelocity` in the player, and removing the "speed" field from the event?

[the remark this answers is not in the document]
Comment (Gocnak) [reply mp/src/game/shared/momentum/mom_gamemovement.cpp:2285]: Nah I mean this entire block can move into the player.

[the remark this answers is not in the document]
Comment (braem) [reply mp/src/game/shared/momentum/mom_gamemovement.cpp:2285]: Ah, gotcha :)

**Comment by braem** [comment]
Made the requested changes, and squashed the logic/events commits into one.

**Comment by braem** [comment]
Here's what it looks like with comparison coloring:
https://cdn.discordapp.com/attachments/549657950201970688/718852675067838544/2020-06-06_08-41-47.mp4

Here's with comparison (separate) coloring:
https://cdn.discordapp.com/attachments/549657950201970688/718852710551781549/2020-06-06_08-42-35.mp4

I prefer with the separate comparison colorization mdoe, but I also prefer last jump velocity with that too so idk if it should be default.

Also: on staged maps, I feel like it should reset the comparison (or not show it) when you get to the next stage. I'll work on something nicer for that case.

**Comment by braem** [comment]
Separated out into 2 labels. Made a new commit for this in case we decide to go back. Vid:
https://cdn.discordapp.com/attachments/549657950201970688/719233711530573954/2020-06-07_09-57-02.mp4

Settings:
image

I defaulted to no colorization for these 2. Also push the fadeout to 10 seconds. I think they wanted no fadeout, with the reasoning being people want to see their speeds, so I just put it to a high value. Can easily remove if you want.

File: mp/src/game/client/momentum/c_mom_player.h [hunk 2]
```diff
@@ -32,6 +32,10 @@ class C_MomentumPlayer : public C_BasePlayer, public CMomRunEntity
     bool HasAutoBhop() { return m_bAutoBhop; }
     // void ResetStrafeSync();
 
+    // Ramp stuff
+    void SetRampBoardVelocity(Vector vecVel);
+    void SetRampLeaveVelocity(Vector vecVel);
```
Comment (Gocnak) [inline mp/src/game/client/momentum/c_mom_player.h:37]:
```suggestion
    void SetRampBoardVelocity(const Vector &vecVel);
    void SetRampLeaveVelocity(const Vector &vecVel);
```

File: mp/src/game/client/momentum/c_mom_player.cpp [hunk 3]
```diff
@@ -122,6 +131,32 @@ void C_MomentumPlayer::PostDataUpdate(DataUpdateType_t updateType)
...
+    m_vecRampBoardVel = vecVel;
+    m_bSurfing = true;
+
+    IGameEvent *pEvent = gameeventmanager->CreateEvent("ramp_board");
+    if (pEvent)
+    {
+        pEvent->SetFloat("speed", vecVel.Length());
+        gameeventmanager->FireEvent(pEvent);
+    }
+}
```
Comment (Gocnak) [inline mp/src/game/client/momentum/c_mom_player.cpp:145]: Could this move to mom_player_shared ? See `FireBullet`

Comment (Gocnak) [reply mp/src/game/client/momentum/c_mom_player.cpp:145]: Also, does this event need to be fired on both server and client side?

Comment (braem) [reply mp/src/game/client/momentum/c_mom_player.cpp]: Moved this to `mom_player_shared` as yeah it does the same stuff. Put an `ifdef` around the event firing as only the server needs to fire that yeah.

**Pull request merged**